vars/piperPipelineStageArtifactDeployment.groovy

```diff
@@ -0,0 +1,77 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.DownloadCacheUtils
+import com.sap.piper.GenerateStageDocumentation
+import com.sap.piper.ReportAggregator
+import com.sap.piper.Utils
+import groovy.transform.Field
+
+import static groovy.json.JsonOutput.toJson
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+
+@Field Set GENERAL_CONFIG_KEYS = []
+@Field Set STAGE_STEP_KEYS = [
+    /** Parameters for deployment to a Nexus Repository Manager. */
+    'nexus'
```

Comment:
Why nexus? Why at all a config?

Reply:
I think it should be rather nexusUpload

Reply:
It usually just about whether to run the step within the stage: https://github.com/SAP/jenkins-library/blob/9e539e68eef1f78e04f754d1910ee5246e6ef5c2/vars/piperPipelineStagePromote.groovy#L35

Reply:
I will have another PR soon that addresses the mismatch of config sources between the Groovy and GO side. Then all this can be removed indeed and will work via aliases.

Reply:
Contrary to what I just wrote, I need to do the conversion of additionalClassifiers somewhere. Either here, or in nexusUpload.groovy. The means in one of these places I need to do config assembly, even if only for this one parameter. There is the problem that for backwards compatibility, we need to read it from a "nexus" sub-section of "artifactDeployment". The other option is to introduce a breaking change and split additionalClassifiers into two list of strings. Then the go-side can consume it and there is no need for handling any parameters in Groovy.

Reply:
But still I think STAGE_STEP_KEYS has do be empty